changelog/5712.feature.rst

```diff
@@ -0,0 +1,2 @@
+Now all arguments to ``@pytest.mark.parametrize`` need to be explicitly declared in the function signature or via ``indirect``.
+Previously it was possible to omit an argument if a fixture with the same name existed, which was just an accident of implementation and was not meant to be a part of the API.
```

Comment:
@nicoddemus actually I'm wondering a bit if that shouldn't be something like "Previously it was possible to omit an argument if a fixture with same name was declared in `parametrize` and requested in some fixture above". Example from #5712 suggests it, as far as I can understand. In code posted by @harius fixture `argument` doesn't really exist, does it?

Reply:
Sounds good to me, thanks!

Reply:
Actually when I read it again, I think the existing text is better: the fixture doesn't need to be declared above, it just needs to exist.